File: 数组/108.将有序数组转化为二叉搜索树.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

# 高度平衡二叉树左右两个子树高度差不超过1
class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
        if not nums:
            return
        mid_index = len(nums)//2
        root = nums[mid_index]
        left_node = self.sortedArrayToBST(nums[:mid_index])
        right_node = self.sortedArrayToBST(nums[mid_index + 1:])
        node = TreeNode(root, left_node, right_node)
        return node

File: 数组/108.将有序数组转化为二叉搜索树.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {number[]} nums
 * @return {TreeNode}
 */
var sortedArrayToBST = function (nums) {
  if (nums.length <= 0) return;
  const mid = ~~(nums.length / 2);
  const root = nums[mid];
  const left = sortedArrayToBST(nums.slice(0, mid));
  const right = sortedArrayToBST(nums.slice(mid + 1));
  return new TreeNode(root, left, right);
};
